player/audio.c

```diff
@@ -55,14 +55,16 @@ static void update_speed_filters(struct MPContext *mpctx)
         return;
 
     double speed = mpctx->opts->playback_speed;
-    double resample = mpctx->speed_factor_a;
+    double resample = mpctx->speed_factor_a * mpctx->opts->playback_pitch;
     double drop = 1.0;
 
     if (!mpctx->opts->pitch_correction) {
         resample *= speed;
         speed = 1.0;
     }
 
+    speed /= mpctx->opts->playback_pitch;
```

Comment:
maybe contain both adjustments in one place?

```suggestion
    if (mpctx->opts->playback_pitch != 1.0) {
        resample *= mpctx->opts->playback_pitch;
        speed /= mpctx->opts->playback_pitch;
    }
```

Reply:
The way it's currently set up, it works both with or without `--audio-pitch-correction`, so resample has to be multiplied before, and speed divided after.

Reply:
> so resample has to be multiplied before the pitch_correction check

What is the difference between
``` c
double resample = mpctx->speed_factor_a * mpctx->opts->playback_pitch;
if (!mpctx->opts->pitch_correction)
    resample *= speed;
```
and
``` c
double resample = mpctx->speed_factor_a;
if (!mpctx->opts->pitch_correction)
    resample *= speed;
resample *= mpctx->opts->playback_pitch
```
?

Reply:
I think both would work, though the 2nd one has an extra line of code. I'm not sure what the point is of separating it into two distinct statements.

Reply:
> EDIT: I guess it's a bit more self-explanatory when they're together, a comment could be added as well to explain what's happening. I'll look into it.

This was my idea, but no need to change if you don't feel like it, it is fine either way.